tools/accuracy_checker/accuracy_checker/metrics/regression.py

```diff
@@ -110,19 +110,37 @@ def _calculate_diff_regression_rep(self, annotation, prediction):
                 return self.value_differ(next(iter(annotation.value.values())), prediction.value)
             diff_dict = OrderedDict()
             for key in annotation.value:
-                diff = self.value_differ(annotation.value[key], prediction.value[key])
+                a_val = annotation.value[key]
+                p_val = prediction.value[key]
+                a_val = a_val.astype(float) if not np.isscalar(a_val) and np.issubdtype(a_val.dtype,
+                                                                                        np.integer) else a_val
+                p_val = p_val.astype(float) if not np.isscalar(p_val) and np.issubdtype(p_val.dtype,
+                                                                                        np.integer) else p_val
+                diff = self.value_differ(a_val, p_val)
                 if np.ndim(diff) > 1:
                     diff = np.mean(diff)
                 diff_dict[key] = diff
             return diff_dict
         if isinstance(prediction.value, dict):
             if len(prediction.value) != 1:
                 raise ConfigError('annotation for all predictions should be provided')
-            diff = self.value_differ(annotation.value, next(iter(prediction.value.values())))
+            a_val = annotation.value
+            a_val = a_val.astype(float) if not np.isscalar(a_val) and np.issubdtype(a_val.dtype,
```

Comment:
could you please create helper-function to_float for this case to avoid code duplication?